x-pack/plugins/ml/public/application/util/index_utils.ts

```diff
@@ -18,7 +18,25 @@ let indexPatternsContract: DataViewsContract | null = null;
 export async function loadIndexPatterns(indexPatterns: DataViewsContract) {
   indexPatternsContract = indexPatterns;
   const dataViewsContract = getDataViews();
-  indexPatternCache = await dataViewsContract.find('*', 10000);
+  const idsAndTitles = await dataViewsContract.getIdsWithTitle();
```

Comment:
We now load all of the ids and titles of the data views and attempt to get the fields for them to work out if the index really exists.
Once we have this filtered list of ids for data views which do exist, we can load the actual data views.

It might be better to use an exists check rather than `getFieldsForIndexPattern` but that will add additional dependencies on this function.